ipmasq: Add default nonMasq CIDRs if config is empty

brb opened the pull request:
To align better with the vanilla ip-masq-agent, add CIDRs from private IPv4 addr space if ip-masq-agent config is empty or missing.

Also, add a support for `masqLinkLocal` which controls whether masq'ing to the link local cidr should be enabled.

Depends on https://github.com/cilium/cilium/pull/11387 to be merged first.

brb committed:
ipmasq: Add default nonMasq CIDRs if config is empty

To align with the vanilla ip-masq-agent, add CIDRs from private IPv4
addr space if ip-masq-agent config is empty.

Also, add a support for "masqLinkLocal" which controls whether masq'ing
to link local cidr should be enabled.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

aanm commented on pkg/ipmasq/ipmasq.go:
```diff
@@ -33,6 +33,22 @@ import (
 	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "ipmasq")
+
+	defaultNonMasqCIDRs = map[string]net.IPNet{
+		"10.0.0.0/8":      mustParseCIDR("10.0.0.0/8"),
+		"172.16.0.0/12":   mustParseCIDR("172.16.0.0/12"),
+		"192.168.0.0/16":  mustParseCIDR("192.168.0.0/16"),
+		"100.64.0.0/10":   mustParseCIDR("100.64.0.0/10"),
+		"192.0.0.0/24":    mustParseCIDR("192.0.0.0/24"),
+		"192.0.2.0/24":    mustParseCIDR("192.0.2.0/24"),
+		"192.88.99.0/24":  mustParseCIDR("192.88.99.0/24"),
+		"198.18.0.0/15":   mustParseCIDR("198.18.0.0/15"),
+		"198.51.100.0/24": mustParseCIDR("198.51.100.0/24"),
+		"203.0.113.0/24":  mustParseCIDR("203.0.113.0/24"),
+		"240.0.0.0/4":     mustParseCIDR("240.0.0.0/4"),
```
it would be nice to have a comment where these CIDRs came from. Also some documentation updates WRT would be helpful for users.

brb replied:
Added a comment about the CIDRs origin. The documentation will come in a separate PR.

brb committed:
helm: Get rid of unused ipMasqAgent.syncPeriod

The Cilium's ip-masq-agent has switched to fsnotify instead of polling,
so the option is no longer used.

Signed-off-by: Martynas Pumputis <m@lambda.lt>
